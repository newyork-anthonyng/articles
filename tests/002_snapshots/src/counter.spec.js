import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Counter from './counter';

describe('Without snapshots ðŸ˜¥', () => {
  it('should render an h1 tag', () => {
    const wrapper = shallow(
      <Counter />
    );

    expect(wrapper.find('h1').length).toEqual(1);
    expect(wrapper.find('h1').at(0).html()).toEqual('<h1>0</h1>');
  });

  it('should render a button', () => {
    const wrapper = shallow(
      <Counter />
    );

    expect(wrapper.find('button').length).toEqual(1);
  });

  it('should update text when clicking on button', () => {
    const wrapper = shallow(
      <Counter />
    );

    const button = wrapper.find('button');
    button.simulate('click');

    expect(wrapper.find('h1').html()).toEqual('<h1>1</h1>');
  });
});

describe('With snapshots ðŸŽ‰', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Counter />
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should update text when clicking on button', () => {
    const wrapper = shallow(
      <Counter />
    );

    const button = wrapper.find('button');
    button.simulate('click');

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
