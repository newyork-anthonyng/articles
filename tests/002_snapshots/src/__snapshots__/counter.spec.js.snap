// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With snapshots ðŸŽ‰ should render correctly 1`] = `
<div>
  <h1>
    0
  </h1>
  <p>
    Even
  </p>
  <button
    onClick={[Function]}
  >
    Increase
  </button>
</div>
`;

exports[`With snapshots ðŸŽ‰ should update text when clicking on button 1`] = `
<div>
  <h1>
    1
  </h1>
  <p>
    Odd
  </p>
  <button
    onClick={[Function]}
  >
    Increase
  </button>
</div>
`;
